src/storage.rs

```diff
@@ -63,64 +49,88 @@ pub trait Storage {
     fn set_language(&mut self, val: String);
     fn set_option(&mut self, val: String);
     fn set_file(&mut self, val: String);
-    fn save(&self);
-    fn delete(&self);
+    fn save(&mut self) -> Result<()>;
+    fn delete(&mut self) -> Result<()>;
 }
 
 impl Storage for FileStorage {
-    fn get_username(&self) -> String {
-        self.username.to_string()
+    fn get_username(&self) -> Option<&str> {
+        match &self.username {
+            Some(val) => Some(val),
+            None => None,
+        }
```

Comment:
This could also be written like this:

```rust
self.username.as_deref()
```

([`as_deref`](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref) takes a reference of the value inside the option)